jbehave-support-core/src/main/java/org/jbehavesupport/core/rest/RestLoggingInterceptor.java

```diff
@@ -0,0 +1,28 @@
+package org.jbehavesupport.core.rest;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+@Slf4j
+public class RestLoggingInterceptor implements ClientHttpRequestInterceptor {
+
+    @Override
+    public ClientHttpResponse intercept(HttpRequest req, byte[] reqBody, ClientHttpRequestExecution ex) throws IOException {
+        log.trace("Request body: {} with headers: {}", new String(reqBody, StandardCharsets.UTF_8), req.getHeaders());
```

Comment:
Can we put something like if(log.isTraceEnabled()) to avoid unnecessary processing? Or add this interceptor only based on some property or if the log level is trace?